themebuilder/modules/modules-handler.js

```diff
@@ -0,0 +1,96 @@
+class ModulesHandler {
+    constructor(widgetsList) {
+        this.SPECIAL_COMMENT = "tb_widgets_list";
+        this.FILE_FOR_PARSING = "theme.less";
+
+        this.widgets = (widgetsList || []).map(w => w.toLowerCase());
+
+        this.bundledWidgets = [];
+    }
+
+    getWidgetFromImport(importString) {
+        const lastSlashIndex = importString.lastIndexOf("/");
+        const fileName = importString.substr(lastSlashIndex + 1);
+        const dotIndex = fileName.indexOf(".");
+        return fileName.substr(0, dotIndex);
+    }
+
+    availableWidgets(less) {
+        const widgets = [];
+        const widgetsListIndex = less.indexOf(this.SPECIAL_COMMENT);
+
+        if(widgetsListIndex < 0) {
+            return widgets;
+        }
+
+        less
+            .substr(widgetsListIndex + this.SPECIAL_COMMENT.length)
+            .split("\n")
+            .filter(item => !!item)
+            .forEach((importString) => {
+                const lastSlashIndex = importString.lastIndexOf("/");
```

Comment:
Do you suppose to use `getWidgetFromImport` method here?

Reply:
great!